reference-implementation/lib/readable-stream.js

```diff
@@ -592,6 +595,13 @@ function ReadableStreamTee(stream, cloneForBranch2) {
         ReadableStreamDefaultControllerEnqueue(branch2._readableStreamController, value2);
       }
     });
+
+    // If pullPromise rejects with an AssertionError, but the branch is already closed or errored,
+    // the rejection will be silently ignored by ReadableStreamDefaultControllerError.
+    // We have to manually rethrow any AssertionError to make sure they are not ignored.
+    pullPromise.catch(rethrowAssertionErrorRejection);
+
+    return pullPromise;
```

Comment:
Technically speaking, we could also just return `Promise.resolve()` here. We already have a `catch` handler on `reader._closedPromise` to error the branches when the original stream errors, so we don't really *need* to return any rejections from `pull()`. Then it would look more like:

```js
function pullAlgorithm() {
  if (pulling === false) {
    pulling = true;
    ReadableStreamDefaultReaderRead(reader).then(result => {
      pulling = false;
      // ...
    }).catch(rethrowAssertionErrorRejection);
  }
  return Promise.resolve();
}
```
Thoughts?

Reply:
I like the idea of always returning Promise.resolve() as it makes the behaviour more symmetrical.

I don't think we can wrap the whole function in another if(), as it will increase the number of levels of indentation, which is bad for the readability of the standard text. We still have to do an early exit.

Of course, we still need abundant tests of error conditions to be confident it works.

Reply:
Okay, so then it'd become:
```js
function pullAlgorithm() {
  if (pulling === true) {
    return Promise.resolve();
  }
  pulling = true;
  ReadableStreamDefaultReaderRead(reader).then(result => {
    pulling = false;
    // ...
  }).catch(rethrowAssertionErrorRejection);
  return Promise.resolve();
}
```
I guess it's not too bad? 😛